torch/_decomp/decompositions.py

```diff
@@ -4497,6 +4497,20 @@ def take(self, index):
     return flattened[index]
 
 
+@register_decomposition(aten.put)
+@out_wrapper()
+def put(self, index, source, accumulate=False):
+    stride = utils.compute_elementwise_output_strides(self)
+    flattened = self.flatten()
+    flattened = torch.index_put(
+        flattened, [index], source.reshape(index.shape), accumulate
+    )
+    output = flattened.reshape(self.shape)
+    if stride != output.stride():
+        output = prims.copy_strided(output, stride)
```

Comment:
I'm not sure about this part. I get the feeling that this may force an unwanted copy if it can't be fused with the next operation? What happens if you perform a put into an operation that cannot be fused into, like an `mm` or something that has a fallback?

Reply:
When using `put` op as input to `mm` and input strides do not match output strides, there will be an additional triton kernel call of copying the put result prior calling `mm`.

This `if` block is now removed as the function is moved to `torch/_inductor/decompositons.py`